packages/e2e-tests/puppeteer/scripts/web3torrent.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import {Page} from 'puppeteer';
+
+export async function uploadFile(page: Page): Promise<string> {
+  await page.waitForSelector('input[type=file]');
+  const fileToUpload = 'file.pdf';
```

Comment:
Is this just the name of a file, or actually the file contents itself? If the latter, it seems as if the download might finish extremely quickly. Depending on what the test expects to happen, that might lead to problems?

Reply:
I understand that it must be the name of a file. I had to make `file.pdf` in the `e2e-test` directory to get this to work. Recommend that you check this file in.

I think the point still stands that the test behaviour will generally depend on the file size, as well as the speed of the internet connection of the machine the test is run on.

Reply:
I added a step in CI to create a 4MB file with random data for seeding.

Reply:
OK. Nitpick follows!

I'm curious as to the advantages of that approach versus checking in a file. The downsides I can see are that this test won't run out of the box in local development, and that we are introducing some indeterminism into our tests which may not be desirable. (Highly likely that the random data doesn't matter, but then why make it random in the first place?).

Reply:
My thinking was to reduce the git commit size. But I agree. It's not worth the hassle to create the file locally by hand. I added the file in the latest git commit.

Reply:
Can you programmatically generate a random 4MB file (that doesn't get checked in?)

Reply:
That's what I did. @geoknee suggests we commit it in the git so that when you run the tests locally you don't have to remember generating the file. 4M is not that big anyway.

See https://github.com/statechannels/monorepo/pull/1329/files#diff-9ed3cedce43d8747b5dc9942f0b66cedR6

Reply:
I'll just note that we can have all of

- deterministic tests
- run out of the box
- no large files checked in

We just need to avoid 1) generating something *random*, 2) relying on something outside test script such as a command on circle. So perhaps a deterministic generation of a file (or Buffer?) within the jest test https://nodejs.org/api/buffer.html#buffer_class_method_buffer_alloc_size_fill_encoding

Reply:
In the latest commit, I got rid of the checked-in large file and programmatically create a `/tmp/web3torrent-test` instead. `/tmp/` is usually RAM backed partition so it should be fast to read and write data there without hitting the disk.